src/ucp/core/ucp_mm.c

```diff
@@ -112,13 +115,23 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
                 continue;
             }
 
+            base_address = address;
+            reg_length   = length;
+
+            if ((context->config.ext.mem_type_reg_whole_alloc) &&
+                (mem_type == UCS_MEMORY_TYPE_CUDA)) {
```

Comment:
can we make it non-cuda-specific?
for example, configure for which memory types this will be enabled?